From 511a305c504e4c0928eda670598528309da3436d Mon Sep 17 00:00:00 2001
From: Sizhe Zhao <prc.zhao@outlook.com>
Date: Tue, 31 Mar 2026 20:46:50 +0800
Subject: [PATCH] fix(policy): use PK_ACTION_CONFIG for
 set{ZoneSettings2,PolicySettings}

Reference: https://access.redhat.com/security/cve/cve-2026-4948

Upstream: https://github.com/firewalld/firewalld/commit/5fb3914ad830feff6cb2b0670457c60a323c6c6c
CVE: CVE-2026-4948
[Joachim: backport to 2.0.2]
Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/firewall/server/firewalld.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/firewall/server/firewalld.py b/src/firewall/server/firewalld.py
index be53fbab..ceaa4322 100644
--- a/src/firewall/server/firewalld.py
+++ b/src/firewall/server/firewalld.py
@@ -946,7 +946,7 @@ class FirewallD(DbusServiceObject):
         log.debug1("getZoneSettings2(%s)", zone)
         return self.fw.zone.get_config_with_settings_dict(zone)
 
-    @dbus_polkit_require_auth(config.dbus.PK_ACTION_CONFIG_INFO)
+    @dbus_polkit_require_auth(config.dbus.PK_ACTION_CONFIG)
     @dbus_service_method(config.dbus.DBUS_INTERFACE_ZONE, in_signature='sa{sv}')
     @dbus_handle_exceptions
     def setZoneSettings2(self, zone, settings, sender=None):
@@ -970,7 +970,7 @@ class FirewallD(DbusServiceObject):
         log.debug1("policy.getPolicySettings(%s)", policy)
         return self.fw.policy.get_config_with_settings_dict(policy)
 
-    @dbus_polkit_require_auth(config.dbus.PK_ACTION_CONFIG_INFO)
+    @dbus_polkit_require_auth(config.dbus.PK_ACTION_CONFIG)
     @dbus_service_method(config.dbus.DBUS_INTERFACE_POLICY, in_signature='sa{sv}')
     @dbus_handle_exceptions
     def setPolicySettings(self, policy, settings, sender=None):
-- 
2.43.0

